packages/server/src/plugin/subscriptionCallback/index.ts

```diff
@@ -34,10 +34,19 @@ export function ApolloServerPluginSubscriptionCallback(
       // The GA version of callback protocol use camelCase, this is to keep backward compatibility
       callbackUrl = callbackUrl || subscriptionExtension.callback_url;
       id = id || subscriptionExtension.subscription_id;
-      heartbeatIntervalMs =
-        heartbeatIntervalMs ||
-        subscriptionExtension.heartbeat_interval_ms ||
-        5000;
+      if (
+        heartbeatIntervalMs == null &&
+        subscriptionExtension.heartbeat_interval_ms != null
+      ) {
+        // Deprecated field name
+        heartbeatIntervalMs = subscriptionExtension.heartbeat_interval_ms;
+      } else if (
+        heartbeatIntervalMs == null &&
+        subscriptionExtension.heartbeat_interval_ms == null
+      ) {
+        // Default value
+        heartbeatIntervalMs = 5000;
+      }
```

Comment:
This is just handling the `0` is falsey bug, yeah? Nullish coalesce handles this well (0 is preserved):
```ts
      heartbeatIntervalMs =
        heartbeatIntervalMs ??
        subscriptionExtension.heartbeat_interval_ms ??
        5000;
```

`??` is generally a more correct replacement for the commonly used `||` fallback in JS.